src/main/java/io/jenkins/plugins/opentelemetry/job/MonitoringRunListener.java

```diff
@@ -104,42 +125,75 @@ public void postConstruct() {
         runLaunchedCounter =
-                meter.counterBuilder(JenkinsSemanticMetrics.CI_PIPELINE_RUN_LAUNCHED)
-                        .setDescription("Job launched")
-                        .setUnit("{jobs}")
-                        .build();
+            meter.counterBuilder(JenkinsSemanticMetrics.CI_PIPELINE_RUN_LAUNCHED)
+                .setDescription("Job launched")
+                .setUnit("{jobs}")
+                .build();
         runStartedCounter =
-                meter.counterBuilder(JenkinsSemanticMetrics.CI_PIPELINE_RUN_STARTED)
-                        .setDescription("Job started")
-                        .setUnit("{jobs}")
-                        .build();
+            meter.counterBuilder(JenkinsSemanticMetrics.CI_PIPELINE_RUN_STARTED)
+                .setDescription("Job started")
+                .setUnit("{jobs}")
+                .build();
         runSuccessCounter =
-                meter.counterBuilder(JenkinsSemanticMetrics.CI_PIPELINE_RUN_SUCCESS)
-                    .setDescription("Job succeed")
-                    .setUnit("{jobs}")
-                    .build();
+            meter.counterBuilder(JenkinsSemanticMetrics.CI_PIPELINE_RUN_SUCCESS)
+                .setDescription("Job succeed")
+                .setUnit("{jobs}")
+                .build();
         runFailedCounter =
             meter.counterBuilder(JenkinsSemanticMetrics.CI_PIPELINE_RUN_FAILED)
                 .setDescription("Job failed")
                 .setUnit("{jobs}")
                 .build();
         runAbortedCounter =
-                    meter.counterBuilder(JenkinsSemanticMetrics.CI_PIPELINE_RUN_ABORTED)
-                        .setDescription("Job aborted")
-                        .setUnit("{jobs}")
-                        .build();
+            meter.counterBuilder(JenkinsSemanticMetrics.CI_PIPELINE_RUN_ABORTED)
+                .setDescription("Job aborted")
+                .setUnit("{jobs}")
+                .build();
         runCompletedCounter =
-                meter.counterBuilder(JenkinsSemanticMetrics.CI_PIPELINE_RUN_COMPLETED)
-                        .setDescription("Job completed")
-                        .setUnit("{jobs}")
-                        .build();
+            meter.counterBuilder(JenkinsSemanticMetrics.CI_PIPELINE_RUN_COMPLETED)
+                .setDescription("Job completed")
+                .setUnit("{jobs}")
+                .build();
+    }
 
+    @Override
+    public void afterConfiguration(ConfigProperties configProperties) {
+        Pattern newRunDurationHistogramAllowList;
+        Pattern newRunDurationHistogramDenyList;
+        try {
+            newRunDurationHistogramAllowList = Optional
+                .ofNullable(configProperties.getString(JenkinsOtelSemanticAttributes.OTEL_INSTRUMENTATION_JENKINS_RUN_DURATION_ALLOW_LIST))
+                .map(Pattern::compile)
+                .orElse(MATCH_NOTHING);
```

Comment:
Does this mean that by default the `ci.pipeline.run.duration` metric will count all job runs under `#other#` until an admin has configured the `otel.instrumentation.jenkins.run.metric.duration.allow_list` property?

Reply:
It's a very good idea. That way all builds are taken into account for the metric while still keeping cardinality under control.
It also allows writing queries to get statistics for all builds on a Jenkins instance (by aggregating away the `ci.pipeline.id`).